### PR TITLE
add workflow for zonal masters and workers in different subnets

### PR DESCRIPTION
**Pull request by WenXinWei**

### CODE REVIEW

**Comment by WenXinWei** [comment]
@gpei @jianlinliu please help review the pr when you have time. thanks

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        networks:
+        - ${LEASED_RESOURCE}
+        datastore: mdcnc-ds-shared
+    - name: us-east-2
+      region: us-east
+      zone: us-east-2a
+      topology:
+        computeCluster: /${vsphere_datacenter}/host/vcs-mdcnc-workload-2
+        networks:
+        - ocp-ci-seg-20
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:71]: This setting is for the `multipsubnets` meaning, right? Once we use it as hard code, we should never add it to prow cluster profile any more, right?

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+       - "us-east-1"
+       - "us-east-2"
+       - "us-east-3"
+compute:
+- name: "worker"
+  replicas: 3
+  platform:
+    vsphere:
+      zones:
+       - "us-west-1"
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:44]: The multisubnets is only for controlPlane ?

File: ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh [hunk 3]
```diff
@@ -10,6 +9,9 @@ if [[ -z "${LEASED_RESOURCE}" ]]; then
   exit 1
 fi
 
+LEASE_NUMBER=$((${LEASED_RESOURCE//[!0-9]/}))
+declare LEASE_NUMBER
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh:13]: I guess we do not need the `declare` line, if we already defined it on the prior line.

File: ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh [hunk 4]
```diff
@@ -165,10 +169,17 @@ curl -sSL "https://mirror2.openshift.com/pub/openshift-v4/clients/butane/latest/
 
 LB_VMNAME="${cluster_name}-lb"
 export GOVC_NETWORK="${LEASED_RESOURCE}"
-govc vm.clone -on=false -vm="${vm_template}" ${LB_VMNAME}
+if [ ${LEASE_NUMBER} -ge 151 ] && [ ${LEASE_NUMBER} -le 157 ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh:172]: the 151-157 segment is for `vsphere-multizone`?

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh (hunk 2 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:44]: yes, refer to https://issues.redhat.com/browse/OCPSTRAT-462.

File: ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh (hunk 4 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh:172]: yes

File: ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh (hunk 3 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh:13]: ok, a mistake. will delete.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-chain.metadata.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+	"path": "ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-chain.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-chain.metadata.json:2]: Suggest to add one more folder `external-lb` to be `ipi/conf/vsphere/zones/multisubnets/external-lb`

File: ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh (hunk 4 above)
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh:172]: Can we add some comment lines to indicated what is the `LEASE_NUMBER` checking for?

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:44]: if we also place workers in multisubnets, what would happen? Does it work?

File: ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh (hunk 4 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/lb/external/ipi-conf-vsphere-lb-external-commands.sh:172]: ok

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh (hunk 2 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:44]: It has not been verified yet, if we need to support worker in multisubnets, I guess we also need apply more subnets like ocp-ci-seg-20 and ocp-ci-seg-21.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh (hunk 1 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:71]: sorry, I'm not very clear about your comments.  do you mean multisubnets maybe also need by workers, so i should not only hard code for master?

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh (hunk 2 above)
Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:44]: I think we do not need more subnets for workers, we can re-use the same subnets for workers, but before that, we need to identify what "multipsubnets" does mean, does it mean masters and workers are placed into diffferent subnets, or we just need to ensure each master in different subnets. 

Personally we can place each worker in different subnets together in this PR, if we do not, I would suggest to rename the step name to `ipi-conf-vsphere-zones-multisubnets-masters` to indicate the step is only for masters.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:71]: In a common case, we should place all the nodes in `${LEASED_RESOURCE}`, this case is for `multsubnets`, so we set a hard-code segment for the domains on purpose, right?

Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:71]: yes, that is.

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:71]: The `ocp-ci-seg-20` and `ocp-ci-seg-21` is newly created by DPP team, dedicated for this case?

Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:71]: we applied for new subnets, but ran in to issue and they suggested to use these two segments.

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

File: ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/zones/multisubnets/external-lb/provision/cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb-provision-chain.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+chain:
+  as: cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb-provision
+  steps:
+  - chain: ipi-conf-vsphere-zones-multisubnets-external-lb
+  - chain: ipi-install-vsphere
+  - ref: enable-qe-catalogsource
+  documentation: |-
```
Comment (gpei) [inline ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/zones/multisubnets/external-lb/provision/cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb-provision-chain.yaml:7]: let's add the `cucushift-installer-check` chain to the provision-chain too

Comment (WenXinWei) [reply ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/zones/multisubnets/external-lb/provision/cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb-provision-chain.yaml:7]: oh, thanks for reminding.

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+ref:
+  as: ipi-conf-vsphere-zones-multisubnets-external-lb
+  from: upi-installer
+  commands: ipi-conf-vsphere-zones-multisubnets-external-lb-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  credentials:
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml:9]: Seem like we do not need these credentials in this step.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.metadata.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{
+	"path": "ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.metadata.json:2]: For this step, sounds not need to indicate any usage of external-lb, so it should be in `ipi/conf/vsphere/zones/multisubnets/`

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 9]
```diff
@@ -3391,6 +3391,16 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-long-duration
     workflow: cucushift-installer-rehearse-vsphere-ipi-zones
+- as: vsphere-ipi-zones-multisubnets-external-lb-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3394]: I think we can use this job to replace `vsphere-ipi-external-lb-pre-f14`

**Comment by jianlinliu** [comment]
Based on we already place both masters and workers in subnets, we should update the PR title accordingly.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.metadata.json (hunk 8 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.metadata.json:2]: ok, got it.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml (hunk 7 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml:9]: ok, but how to judge whether to use these credentials or not?

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml:9]: Searching `/var/run/vault/vsphere` in the commands.sh

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml (hunk 9 above)
Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3394]: ok

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml (hunk 7 above)
Comment (WenXinWei) [reply ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/external-lb/ipi-conf-vsphere-zones-multisubnets-external-lb-ref.yaml:9]: thanks.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 10]
```diff
@@ -3391,6 +3381,16 @@ tests:
...
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3384]: Based on https://github.com/openshift/release/pull/42996 is adding destructive jobs, to keep consistent, pls add one more destructive job for it.

Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3550]: ok

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 11]
```diff
@@ -3557,6 +3547,28 @@ tests:
...
+  cron: 57 20 25 * *
+  steps:
+    cluster_profile: vsphere-multizone
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-ipi and @network-ovnkubernetes and not @fips'
+      TAG_VERSION: '@4.14'
+    test:
+    - chain: openshift-e2e-test-qe-destructive
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3558]: Wrong e2e chain?

Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3558]: sorry, forgot to change.

File: ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+      zone: us-west-1a
+      topology:
+        datacenter: datacenter-2
+        computeCluster: /datacenter-2/host/vcs-mdcnc-workload-4
+        networks:
+        - ${LEASED_RESOURCE}
+        datastore: mdcnc-ds-shared
+
+networking:
+  networkType: OpenShiftSDN
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/vsphere/zones/multisubnets/ipi-conf-vsphere-zones-multisubnets-commands.sh:94]: By a chance, I think we should remove `networkType` setting, to ask installer to decide the default network type.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 13]
```diff
@@ -3557,6 +3547,28 @@ tests:
...
+    workflow: cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb
+- as: vsphere-ipi-zones-multisubnets-external-lb-f28-destructive
+  cron: 51 3 24 * *
+  steps:
+    cluster_profile: vsphere-multizone
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-ipi and @network-ovnkubernetes and not @fips
+        and @destructive'
+      TAG_VERSION: '@4.14'
+      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3568]: For the setting, I think you can refer to https://github.com/openshift/release/pull/42996/files#r1315613753

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.13-e2e-vsphere-externallb-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-vsphere-ipi-zones-multisubnets-external-lb-f28

**Comment by WenXinWei** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn

**Comment by jianlinliu** [comment]
periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn job always failed with "" error, refer to the job's [history](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.14-e2e-vsphere-externallb-ovn/1698927176847462400), also failed with the same error,  so seem like not related with this PR change.

**Comment by jianlinliu** [comment]
cc @liangxia to review

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 14]
```diff
@@ -3691,6 +3691,27 @@ tests:
...
+    test:
+    - chain: openshift-e2e-test-qe
+    workflow: cucushift-installer-rehearse-vsphere-ipi-zones-multisubnets-external-lb
+- as: vsphere-ipi-zones-multisubnets-external-lb-f28-destructive
+  cron: 51 3 24 * *
+  steps:
+    cluster_profile: vsphere-multizone
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-ipi and @network-ovnkubernetes and not @fips
+        and @destructive'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3710]: Would you mind to remove `and @destructive` here, it's no longer needed, refer to https://github.com/openshift/release/pull/43005

Comment (WenXinWei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3710]: no prob.

**Pull request merged**